### PR TITLE
test-case: add echo reference test case

### PR DESCRIPTION
**Pull request by keqiaozhang**
add one new test case to check the echo reference on
on supported platforms.

Signed-off-by: Zhang Keqiao <keqiao.zhang@linux.intel.com>

### CODE REVIEW

**Comment by xiulipan** [comment]
@keqiaozhang Do we really need a new test scripts for the echo reference test? I think this is almost same like `check-alsabat.sh`  
Also please name this as FW echo reference to make difference with the DSM hw echo reference

**Comment by keqiaozhang** [comment]
> @keqiaozhang Do we really need a new test scripts for the echo reference test? I think this is almost same like `check-alsabat.sh`

> Also please name this as FW echo reference to make difference with the DSM hw echo reference

Yes, I considered to reuse alsabat test script before, but we better to use tplgreader to get the PCM IDs of echo reference pipelines instead of the hard code and also in alsabat test script, we use plugin, but we want to test echo reference with different formats.
Case name is updated to check-fw-echo-reference.sh

**Comment by marc-hb** [comment]
13 `shellcheck disable` (with barely any comment) in a single small file like this really doesn't really make sense, fixed in PR #315 .

File: test-case/check-fw-echo-reference.sh [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        if [ "$fmt" == "S24_LE" ]; then
+            continue
+        fi
+        # run echo reference test
+        dlogc "alsabat -P $pb_dev --standalone -c $channel -f $fmt -r $rate -n $frames -F $frequency"
+        alsabat -P "$pb_dev" --standalone -c "$channel" -f "$fmt" -r "$rate" -n "$frames" -F "$frequency" &
+        # playback may have low latency, add 0.5 second delay to aviod recording zero at beginning.
+        sleep 1
+        dlogc "alsabat -C $cp_dev -c $channel -f $fmt -r $rate -F $frequency"
+        alsabat -C "$cp_dev" -c "$channel" -f "$fmt" -r "$rate" -F "$frequency"
```
Comment (marc-hb) [inline test-case/check-fw-echo-reference.sh:104]: Please help review #313 and then test this script with `set -e`, see #312 

I think we don't want to risk missing any `alsabat` failure.

File: test-case/check-fw-echo-reference.sh [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    for fmt in $fmts
+    do
+    printf "Testing: iteration %d of %d with %s format\n" "$i" "$loop_cnt" "$fmt"
+        # S24_LE format is not supported
+        if [ "$fmt" == "S24_LE" ]; then
+            continue
+        fi
+        # run echo reference test
+        dlogc "alsabat -P $pb_dev --standalone -c $channel -f $fmt -r $rate -n $frames -F $frequency"
+        alsabat -P "$pb_dev" --standalone -c "$channel" -f "$fmt" -r "$rate" -n "$frames" -F "$frequency" &
```
Comment (marc-hb) [inline test-case/check-fw-echo-reference.sh:100]: Make sure we don't miss any `alsabat` failure, this is a test script:
```shell
alsabat ... & alsabatPID=$!
...
...
timeout 5 wait $alsabatPID
```

**Pull request merged**